docs/develop/python/user-metadata.mdx

```diff
@@ -0,0 +1,259 @@
+---
+id: user-metadata
+title: User Metadata - Python SDK
+sidebar_label: User Metadata
+description: Learn how to add user metadata to workflows, activities, timers, signals, queries, and updates in the Temporal Python SDK.
+toc_max_heading_level: 3
+keywords:
+  - user metadata
+  - workflow metadata
+  - static summary
+  - static details
+  - activity metadata
+  - timer metadata
+  - signal metadata
+  - query metadata
+  - update metadata
+tags:
+  - User Metadata
+  - Python SDK
+  - Temporal SDKs
+---
+
+# User Metadata
+
+User metadata allows you to attach arbitrary information to workflows and events. This information can be displayed in the Temporal UI, making it easier to identify and understand workflows and their operations.
+
+## Adding Metadata
+
+### Workflow
+
+#### Starting a Workflow
+
+When starting a workflow, you can provide a static summary and details to help identify the workflow in the UI:
+
+```python
+# Start a workflow with static summary and details
+handle = await client.start_workflow(
+    YourWorkflow.run,
+    "workflow input",
+    id="your-workflow-id",
+    task_queue="your-task-queue",
+    static_summary="Order processing for customer #12345",
+    static_details="Processing premium order with expedited shipping"
+)
+```
+
+The `static_summary` is a single-line description that appears in the workflow list view, while `static_details` can be multi-line and provides more comprehensive information that appears in the workflow details view.
+
+You can also use the `execute_workflow` method with the same parameters:
+
+```python
+result = await client.execute_workflow(
+    YourWorkflow.run,
+    "workflow input",
+    id="your-workflow-id",
+    task_queue="your-task-queue",
+    static_summary="Order processing for customer #12345",
+    static_details="Processing premium order with expedited shipping"
+)
+```
+
+#### Inside the Workflow
+
+Within a workflow, you can get and set the _current workflow details_. These details are distinct from the static details provided to workflow start/execute, and are included in what is returned from the `__temporal_workflow_metadata` query.
```

Comment:
This is an internal query and its name is not important. It's just important to explain "Static summary and static details are fixed at workflow start, however, current details can be changed throughout the workflow. Current details are fetched by interfaces like the UI via query."